src/Symfony/Component/Messenger/Tests/Asynchronous/Routing/SenderLocatorTest.php

```diff
@@ -36,6 +37,38 @@ public function testItReturnsTheSenderBasedOnTheMessageClass()
         $this->assertEquals(array(), $locator->getSendersForMessage(new SecondMessage()));
     }
 
+    public function testItReturnsTheSenderBasedOnTheMessageParentClass()
+    {
+        $sender = $this->getMockBuilder(SenderInterface::class)->getMock();
+        $container = new Container();
+        $container->set('my_amqp_sender', $sender);
+
+        $locator = new SenderLocator($container, array(
+            DummyMessage::class => array(
```

Comment:
Can you configure it here so that the interface is the first argument of the locator to prove that the parent class takes precedence on the interface?

Reply:
Done.